src/services/MnemonicSave.ts

```diff
@@ -0,0 +1,48 @@
+import { Storage } from "@utils/StorageArray";
+
+export class MnemonicSave {
+  public mnemonicArr: string[];
+  public shuffledMnemonic: string[];
+  private counter: number;
+  constructor(private readonly storage: Storage) {
+    this.mnemonicArr = [];
+    this.shuffledMnemonic = [];
+    this.counter = 0;
+  }
+
+  async checkNeeded() {
+    return await this.storage.hasItem(this.setupKey());
+  }
+  async addCheckNeeded() {
```

Comment:
"add" implies you could do this multiple times. Like [add, add, remove] would still have checkNeeded. I'd call this `setCheckNeeded` and the other method `setCheckNotNeeded`.